### ISSUE TITLE
Blazor EditorRequired attribute incorrectly reports RZ2012 warning when @bind-MyVariable= is used instead of MyVariable= syntax.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#39229]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Blazor-EditorRequired-attribute-incorrec/1608221)._

---
[severity:It bothers me. A fix would be nice]
In a blazor component, I want to require that a string parameter be provided, but I also want to bind to it like this:

[EditorRequired]
        [Parameter]
        public string GLCode { get; set; } = null!;
     
[Parameter]
        public EventCallback<string> GLCodeChanged { get; set; }

When I use that component, and I provide the value with a @bind-GLCode="@parentGLCode", I get an incorrect RZ2012 warning that GLCode may not have been provided.

<GLCodePickerTwoColumnComponent 
                  @bind-GLCode="@localDistribution.GLAccount">
         </GLCodePickerTwoColumnComponent>

Warning (active)	RZ2012	Component 'GLCodePickerTwoColumnComponent' expects a value for the parameter 'GLCode', but a value may not have been provided.

---
#### Feedback Bot on 12/8/2021, 07:18 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by szalapski** [comment]
This is a bug in the current state of [EditorRequired] and I think it should be fixed as a priority.

**Comment by y2kbugger** [comment]
> This is a bug in the current state of how components with [EditorRequired] build and I think it should be fixed as a priority. I'm surprised EditorRequired got into .NET 6 with such an obvious use case failing. Thanks for your consideration.

I agree, should be considered as a bug fix in a .NET 6 point release
